### PR TITLE
Upgrade google-auth-library.

### LINKED ISSUE #35: Concerns about PR #34
I'm very concerned about PR #34. Specifically I'm worried how the version of google-auth-library to 1.2.0 was updated.

After updating the version of google-auto-auth used in @google-cloud/common (locally) and then depending on @google-cloud/common locally in the module @google-cloud/profiler, several tests failed (due to timeouts) when they had previously passed.

My primary concern is not that I've seen failing tests after this update, though. It seems to me as if PR #34 switched to using version 1.2.0 of google-auth-library, but did not include other notable changes, even though the API of google-auth-library changed notably with the major version update.

Specifically, GoogleAuth.fromJson() no longer takes a callback function. (Current version [here](https://github.com/google/google-auth-library-nodejs/blob/5ab6be36bed5fd2ae0031ae19363270b3a828133/src/auth/googleauth.ts#L382); And code from a version prior to 1.0.0 [here](https://github.com/google/google-auth-library-nodejs/blob/d50aeda72373860504ed84047d5427d059c7c9f8/ts/lib/auth/googleauth.ts#L391)). But  it looks like GoogleAuth.fromJson() is still called with a callback [here](https://github.com/stephenplusplus/google-auto-auth/blob/51137e094fa0ab57d82d83985396a86e905be692/index.js#L89) and [here](https://github.com/stephenplusplus/google-auto-auth/blob/51137e094fa0ab57d82d83985396a86e905be692/index.js#L100).

It is my impression that there were pretty notable changes to the API of google-auth-library with the major version change. And it looks like google-auth-library is mocked in the tests for google-auto-auth, so I'm not confident that the tests would capture problems that might arise due to the changes made in PR#34.

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #35

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -86,7 +86,12 @@ class Auth {
       };
 
       if (config.credentials) {
-        googleAuthClient.fromJSON(config.credentials, addScope);
```
Comment (stephenplusplus) [inline index.js]: @ofrobots previously, I *think* the callback to `fromJSON()` would be executed with:

- err
- auth client instance
- project ID (if detected)

Now that the callback is gone, and the auth client is returned directly from `fromJSON()`, is the project ID being parsed from the provided object and stored somewhere?

**Comment by stephenplusplus** [comment]
@ofrobots could you please take a look at this?

File: index.js (hunk 1 above)
Comment (kjin) [reply index.js]: In the source for `fromJSON` from 0.12.0, it seems like the callback was never called with the project ID: https://github.com/google/google-auth-library-nodejs/blob/v0.12.0/ts/lib/auth/googleauth.ts#L391

I think the only project ID involved is the maybe-`client.projectId` field.

Comment (stephenplusplus) [reply index.js]: Alright, thanks for checking that out!

**Pull request merged**